tests/test/tuples.rs

```diff
@@ -34,3 +34,101 @@ fn tuple_trait_impl() {
         }
     }
 }
+
+#[test]
+fn tuples_are_sized() {
+    test! {
+        program {
+            #[lang(sized)]
+            trait Sized { }
+
+            trait Foo {}
+        }
+
+        goal {
+            (dyn Foo,): Sized
+        } yields {
+            "No possible solution"
+        }
+
+        goal {
+            (u8, dyn Foo): Sized
+        } yields {
+            "No possible solution"
+        }
+
+        goal {
+            (dyn Foo, u8): Sized
+        } yields {
+            "Unique; substitution [], lifetime constraints []"
```

Comment:
probably worth a comment here about how this is not a well-formed type

Reply:
I added:
```
        // It should not be well-formed because for tuples, only
        // the last element is allowed not to be Sized.
```